cub/device/dispatch/dispatch_histogram.cuh

```diff
@@ -175,6 +175,26 @@ template <
     typename    OffsetT>                    ///< Signed integer type for global offsets
 struct DispatchHistogram
 {
+private:
+    template <class T>
+    CUB_RUNTIME_FUNCTION 
+    static T ComputeScale(T lower_level, T upper_level, int bins)
+    {
+      return static_cast<T>(upper_level - lower_level) / bins;
+    }
+
+#if defined(__CUDA_FP16_TYPES_EXIST__)
+    // There are no host versions of arithmetic operations on `__half`, so 
+    // all arithmetic operations on host shall be done on `float`
+    CUB_RUNTIME_FUNCTION 
+    static __half ComputeScale(__half lower_level, __half upper_level, int bins)
+    {
+      return __float2half(
```

Comment:
Question: Should this also check for SM53, or are the conversion intrinsics always necessary here?

Reply:
SM53 introduces arithmetic operations. Conversions are [present](https://godbolt.org/z/57xdqsdev) in SM35.